components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/utils/CmdUtils.java

```diff
@@ -260,12 +277,89 @@ private static String getProjectTempFolderLocation(String projectName) {
         return getProjectDirectoryPath(projectName) + File.separator + CliConstants.TEMP_DIR_NAME;
     }
 
+    /**
+     * Set api definition file
+     *
+     * @param projectName name of the project
+     * @param apiDefinition api definition file local path
+     * @param apiDefinitionURL api definition download URL
+     * @param headers api definition download request headers
+     * @param values api definition download values for request headers
+     */
+    private static void setApiDefinition(String projectName, String apiDefinition, String apiDefinitionURL,
+                                         String headers, String values, boolean insecure) throws IOException {
+        String apiDefinitionsDir = getUserDir() + File.separator + projectName + File.separator +
+                CliConstants.PROJECT_API_DEFINITIONS_DIR + File.separator;
+        String filePath = "";
+        boolean isDownloaded = false;
+        if (!StringUtils.isEmpty(apiDefinition)) {
+            filePath = apiDefinitionsDir + Paths.get(apiDefinition).getFileName();
+            Files.copy(Paths.get(apiDefinition), Paths.get(filePath));
+            isDownloaded = true;
+            logger.debug("Api definition is successfully copied to :" + filePath);
+        }
+        if (!StringUtils.isEmpty(apiDefinitionURL)) {
+            ArrayList<String> headersList = new ArrayList<>();
+            ArrayList<String> valuesList = new ArrayList<>();
+            if (!StringUtils.isBlank(headers) && !StringUtils.isBlank(values)) {
+                headersList = new ArrayList(Arrays.asList(headers.split(",")));
+                valuesList = new ArrayList(Arrays.asList(values.split(",")));
+                if (headersList.size() > 0) {
+                    if (headersList.size() != valuesList.size()) {
+                        throw new CLIRuntimeException("Provided number of header and number of values is different");
+                    }
+                }
+                logger.debug("Request headers are provided.");
+            }
+            isDownloaded = true;
```

Comment:
Shall we move this line after the `filePath = downloadFile(...` line? Just for logical correctness.

Reply:
Fixed. thanks. c048312